wolfCLU/src/crypto/clu_crypto_setup.c

```diff
@@ -0,0 +1,358 @@
+/* clu_crypto_setup.c
+ *
+ * Copyright (C) 2006-2020 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL. (formerly known as CyaSSL)
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+#include <wolfclu/clu_header_main.h>
+#include <wolfclu/clu_optargs.h>
+
+
+int wolfCLU_setup(int argc, char** argv, char action)
+{
+    char     outNameE[256];     /* default outFile for encrypt */
+    char     outNameD[256];     /* default outfile for decrypt */
+    char     inName[256];       /* name of the in File if not provided */
+
+    char*    name = NULL;       /* string of algorithm, mode, keysize */
+    int      alg;               /* algorithm from name */
+    char*    mode = NULL;       /* mode from name */
+    char*    out  = NULL;       /* default output file name */
+    char*    in = inName;       /* default in data */
+    byte*    pwdKey = NULL;     /* password for generating pwdKey */
+    byte*    key = NULL;        /* user set key NOT PWDBASED */
+    byte*    iv = NULL;         /* iv for initial encryption */
+
+
+    int      size       =   0;  /* keysize from name */
+    int      ret        =   0;  /* return variable */
+    int      block      =   0;  /* block size based on algorithm */
+    int      pwdKeyChk  =   0;  /* if a pwdKey has been provided */
+    int      ivCheck    =   0;  /* if the user sets the IV explicitly */
+    int      keyCheck   =   0;  /* if ivCheck is 1 this should be set also */
+    int      inCheck    =   0;  /* if input has been provided */
+    int      outCheck   =   0;  /* if output has been provided */
+    int      i          =   0;  /* loop counter */
+    int      eCheck     =   0;  /* if user is encrypting data */
+    int      dCheck     =   0;  /* if user is decrypting data */
+    int      inputHex   =   0;  /* if user is encrypting hexidecimal stuff */
+    int      keyType    =   0;  /* tells Decrypt which key it will be using
+                                 * 1 = password based key, 2 = user set key
+                                 */
+    int      printOut   =   0;  /* flag to print out key/iv/salt */
```

Comment:
I'd call this verbose, but just personal preference.

Reply:
changed